sentry_sdk/integrations/django/caching.py

```diff
@@ -1,80 +1,152 @@
-                span.set_data(SPANDATA.CACHE_HIT, False)
+            with capture_internal_exceptions():
+                if address is not None:
+                    span.set_data(SPANDATA.NETWORK_PEER_ADDRESS, address)
+
+                if port is not None:
+                    span.set_data(SPANDATA.NETWORK_PEER_PORT, port)
+
+                key = _get_key(args, kwargs)
+                if key != "":
+                    span.set_data(SPANDATA.CACHE_KEY, key)
+
+                item_size = None
+                if is_get_operation:
+                    if value:
+                        item_size = len(str(value))
+                        span.set_data(SPANDATA.CACHE_HIT, True)
+                    else:
+                        span.set_data(SPANDATA.CACHE_HIT, False)
+                else:
+                    try:
+                        # 'set' command
+                        item_size = len(str(args[1]))
+                    except IndexError:
+                        # 'set_many' command
+                        item_size = len(str(args[0]))
+
+                if item_size is not None:
+                    span.set_data(SPANDATA.CACHE_ITEM_SIZE, item_size)
 
             return value
 
     @functools.wraps(original_method)
     def sentry_method(*args, **kwargs):
         # type: (*Any, **Any) -> Any
-        return _instrument_call(cache, method_name, original_method, args, kwargs)
+        return _instrument_call(
+            cache, method_name, original_method, args, kwargs, address, port
+        )
 
     setattr(cache, method_name, sentry_method)
 
 
-def _patch_cache(cache):
-    # type: (CacheHandler) -> None
+def _patch_cache(cache, address=None, port=None):
+    # type: (CacheHandler, Optional[str], Optional[int]) -> None
     if not hasattr(cache, "_sentry_patched"):
         for method_name in METHODS_TO_INSTRUMENT:
-            _patch_cache_method(cache, method_name)
+            _patch_cache_method(cache, method_name, address, port)
         cache._sentry_patched = True
 
 
+def _get_address_port(settings):
+    # type: (dict[str, Any]) -> tuple[Optional[str], Optional[int]]
+    address, port = None, None
+    location = settings.get("LOCATION")
+    # TODO: location can also be an array of locations
+    # see: https://docs.djangoproject.com/en/5.0/topics/cache/#redis
+    if isinstance(location, str):
+        if ":" in location:
+            address, port = location.rsplit(":", 1)
+        else:
+            address, port = location, None
+
+    # If address is a URL (could also be a string identifier or path to a Unix socket)
+    # remove the username and password from URL to not leak sensitive data.
+    if address is not None and "://" in address:
+        parsed_url = urlparse(address)
+        address = "{}://{}{}".format(
+            parsed_url.scheme or "",
+            parsed_url.netloc or "",
+            parsed_url.path or "",
+        )
+
+    return address, int(port) if port is not None else None
```

Comment:
[nitpick] the logic would be clearer to me if we return early with `None, None` when `location` is not a `str`. However, I am also fine with the original code, if you prefer it

```suggestion
def _get_address_port(settings):
    # type: (dict[str, Any]) -> tuple[Optional[str], Optional[int]]
    location = settings.get("LOCATION")
    if not isinstance(location, str):
        return None, None
    
    if ":" in location:  # TODO: Still need to fix this logic
        address, port = location.rsplit(":", 1)
    else:
        address, port = location, None

    # If address is a URL (could also be a string identifier or path to a Unix socket)
    # remove the username and password from URL to not leak sensitive data.
    if "://" in address:
        parsed_url = urlparse(address)
        address = "{}://{}{}".format(
            parsed_url.scheme or "",
            parsed_url.netloc or "",
            parsed_url.path or "",
        )

    return address, int(port) if port is not None else None
```

Reply:
done